app/controllers/resources_controller.rb

```diff
@@ -1,14 +1,22 @@
 class ResourcesController < ApplicationController
   def index
     category_id = params[:category_id]
+    latitude = params[:latitude]
+    longitude = params[:longitude]
+
     result = if category_id
                resources.joins(:categories).where('categories.id' => category_id)
              end
 
     if result
+      if latitude and longitude
+        result = result.sort_by{ |resource| resource.address.distance_to([latitude,longitude])}
```

Comment:
Postgres actually comes with an extension for computing distances on the surface of the earth. We just need to install it (with a migration). http://tapoueh.org/blog/2013/08/05-earthdistance has an example of its use. We could then do this entirely in the query above with something like:

```ruby
resources.joins(:categories).joins(:address).
  where('categories.id' => category_id).
  order('point(?, ?) <@> point(address.longitude, address.latitude)',
    longitude, latitude)
```